### DISCUSSION TITLE
Ruby 2 support plans

### DISCUSSION THREAD

**Discussion by timsutton** [fastlane/fastlane#21171] [General]
Hello! Long-time Fastlane fan :D

I wondered whether there are any specific plans/timelines for deprecating support for Ruby 2.x? It's EOL at the end of March 2023. Fastlane is widely-used so I imagine deprecating Ruby 2 might be still a ways out, and on the other hand it has a lot of dependencies of its own. Also perhaps there are users using Apple's old system Ruby 2.6 with it?

Anyway, was just interested if maintainers have any thoughts or plans around this. I saw [this one comment](https://github.com/fastlane/fastlane/discussions/20463#discussioncomment-3751737) from @rogerluan in the 3.0 ideas thread, but 3.0 looks like there's a lot of ideas and I'm unclear if there's any concrete plans around timing there.

Thanks for the consideration and everyone's maintainership on Fastlane! 🙏

**Comment by rogerluan** [comment]
Hi @timsutton 👋 

Thanks for bringing this up to our attention 🙏  There hasn't been any official conversations about deprecating Ruby 2.x (to my knowledge), but with that being said, we could use this space to start discussing those changes 😊 

_fastlane_ is used by thousands and thousands of indie developers and companies globally, so we always have to move forward with a lot of care and attention when the change is not backwards compatible (including dropping support to a Ruby version).

May I ask why you're bringing this up, and how this is relevant to you? fastlane has already been supporting Ruby 3.0 and 3.1 for a long time now, I wonder if you have any interest in specifically seeing support to Ruby 2.x being dropped?

**Reply by timsutton** [reply]
Thanks for the quick response! That all makes sense to me.

> May I ask why you're bringing this up, and how this is relevant to you? fastlane has already been supporting Ruby 3.0 and 3.1 for a long time now, I wonder if you have any interest in specifically seeing support to Ruby 2.x being dropped?

Internally we have a lot of Ruby code in terms of CI tooling, and Fastlane is one part of that. The majority of our tools haven't yet been updated to be 3.x compatible but now with the 2.7 EOL I'm feeling like we're starting to be under the gun.

So the reason I was interested, was because if for example I knew that Fastlane was already planning to drop 2.7 support from the time of its EOL, then it would help us prioritize when exactly we would migrate all our code, or at least _which_ code we'd migrate first. So nothing blocks us to migrate to 3.0/3.1 except our own time :)

**Reply by rogerluan** [reply]
Got it! If that's the case, then I think I can say there are no plans (again: AFAIK haha) to drop support to 2.x, for the time being.

And you can upgrade your fastlane to use Ruby 3.x already, so whenever your team/project is ready, fastlane already works with it 😊
